BCIF byteArrayEncoder type fix

epeisach opened the pull request:
For byteArrayEncoder, if array type is specified, then it should be used.
Provide warnings that old encoders without array types will be deprecated in 2026.

Have tested encoding PDB archive and used java based decoder to validate.

piehld reviewed (commented):
Thanks @epeisach! The only thing I'd recommend is adding one of the problematic structures to the tests (could just pull it from our archive via URL, rather than putting the file in the repository).

epeisach commented:
I like your idea.  I will find a smaller example --- not 3j3q...

However, the issue is that we are not generating a valid output.  The parser is not strictly checking the inputs - and would need a java runtime....   With that said - I will think about how to write a simple parser that would detect failure...

epeisach commented:
I have added my binary cif decoder to the tests - and added verification of the encoding.  I then integrated with test suite and added 8ccs as a test example to use.

piehld commented on mmcif/tests/testBinaryCifWriter.py:
```diff
@@ -129,6 +134,18 @@ def testSerialize(self):
+    def __verifyEncoding(self, fname, storeStringsAsBytes):
+        """Verifies encoding"""
+        exe = sys.executable
+        args = [exe, os.path.join(HERE, "bcifprint.py"), fname]
```
Why choose to call `bcifprint.py` as an executable file, rather than importing the class and calling some method? Is it because you want to be able to use that script for other cases of testing?

epeisach replied:
I can import it....  I can rewrite....

piehld replied:
I mean, if you think that script is useful on its own, then don't worry about it. But if not, then yeah it would be good to import it instead (or just stick it in the `testBinaryCifWriter.py` directly), to be consistent with all of our other package tests.

epeisach replied:
I have left the script around as a standalone - but modified it to suppress output.  That is now imported by the testing.

piehld reviewed (commented):
Thanks @epeisach, changes look great to me. My only remaining nitpick is if you can change the filename `bcifprint.py` to `BcifPrint.py` (and the corresponding import), to be consistent with the class name?

Otherwise, looks good to me!

epeisach commented:
Done.   Tests are running.

piehld reviewed (approved):
Thanks @epeisach!
